compiler/rustc_middle/src/ty/error.rs

```diff
@@ -985,6 +990,38 @@ fn foo(&self) -> Self::T { String::new() }
         false
     }
 
+    pub fn short_ty_string(self, ty: Ty<'tcx>) -> Result<String, (String, PathBuf)> {
```

Comment:
Considering both callers to this function convert the result into an `(String, Option<PathBuf>)` maybe make that the return type of this function?